### DISCUSSION TITLE
Error in compiling LDT

### DISCUSSION THREAD

**Discussion by srikar-iist** [NASA-LIS/LISF#1201] [Q&A] [answered]
Hi, I have sucessfully installed all dependencies for LDT and have successfully configured LDT.

Enable GeoTIFF support? (1-yes, 0-no, default=1): 0
Enable LIBGEOTIFF support? (1-yes, 0-no, default=1): 0
Include date/time stamp history? (1-yes, 0-no, default=1): 0
-----------------------------------------------------
 **configure.ldt file generated successfully**
-----------------------------------------------------
Settings are written to configure.ldt in the make directory.
If you wish to change settings, please edit that file.

To compile, run the compile script.

But the LDT compilation shows the following error:

mpiifort -c -O2 -u -traceback -fpe0  -nomixed_str_len_arg -names lowercase -convert big_endian -assume byterecl  -DSPMD -DIFC -I/projects/iist01/parent/LIS/depend/esmf/esmf/mod/modO/Linux.intel.64.intelmpi.default -DUSE_INCLUDE_MPI -I/projects/iist01/parent/LIS/depend/include/ -I/projects/iist01/parent/LIS/depend/include/  -I/projects/iist01/parent/LIS/depend/include/  -I/projects/iist01/parent/LIS/depend/include/  -I/projects/iist01/parent/LIS/depend/include/  -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../core/LDT_historyMod.F90
../core/LDT_historyMod.F90(56): error #7013: This module file was not generated by any release of this compiler.   [NETCDF]
  use netcdf
------^
../core/LDT_historyMod.F90(2564): error #7013: This module file was not generated by any release of this compiler.   [NETCDF]
    use netcdf
--------^
../core/LDT_historyMod.F90(2761): error #7013: This module file was not generated by any release of this compiler.   [NETCDF]
    use netcdf
--------^
../core/LDT_historyMod.F90(327): error #6404: This name does not have a type, and must have an explicit type.   [NF90_DEF_DIM]
          call LDT_verify(nf90_def_dim(ftn,'east_west',LDT_rc%gnc(n),dimID(1)))
--------------------------^
../core/LDT_historyMod.F90(327): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
          call LDT_verify(nf90_def_dim(ftn,'east_west',LDT_rc%gnc(n),dimID(1)))
---------------^
../core/LDT_historyMod.F90(328): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
          call LDT_verify(nf90_def_dim(ftn,'north_south',LDT_rc%gnr(n),dimID(2)))
---------------^
../core/LDT_historyMod.F90(329): error #6404: This name does not have a type, and must have an explicit type.   [NF90_GLOBAL]
          call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"missing_value", -9999.0))          
-------------------------------------------^
../core/LDT_historyMod.F90(329): error #6404: This name does not have a type, and must have an explicit type.   [NF90_PUT_ATT]
          call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"missing_value", -9999.0))          
--------------------------^
../core/LDT_historyMod.F90(329): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
          call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"missing_value", -9999.0))          
---------------^
../core/LDT_historyMod.F90(333): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"MAP_PROJECTION", &
------------------^
../core/LDT_historyMod.F90(335): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"SOUTH_EAST_CORNER_LAT", &
------------------^
../core/LDT_historyMod.F90(337): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"SOUTH_EAST_CORNER_LON", &
------------------^
../core/LDT_historyMod.F90(339): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"DX", &
------------------^
../core/LDT_historyMod.F90(341): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"DY", &
------------------^
../core/LDT_historyMod.F90(345): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"MAP_PROJECTION", &
------------------^
../core/LDT_historyMod.F90(347): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"SOUTH_EAST_CORNER_LAT", &
------------------^
../core/LDT_historyMod.F90(349): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"SOUTH_EAST_CORNER_LON", &
------------------^
../core/LDT_historyMod.F90(351): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"TRUELAT1", &
------------------^
../core/LDT_historyMod.F90(353): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"STANDARD_LON", &
------------------^
../core/LDT_historyMod.F90(355): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"DX", &
------------------^
../core/LDT_historyMod.F90(357): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"DY", &
------------------^
../core/LDT_historyMod.F90(361): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"MAP_PROJECTION", &
------------------^
../core/LDT_historyMod.F90(363): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"SOUTH_EAST_CORNER_LAT", &
------------------^
../core/LDT_historyMod.F90(365): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"SOUTH_EAST_CORNER_LON", &
------------------^
../core/LDT_historyMod.F90(367): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"TRUELAT1", &
------------------^
../core/LDT_historyMod.F90(369): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"TRUELAT2", &
------------------^
../core/LDT_historyMod.F90(371): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"STANDARD_LON", &
------------------^
../core/LDT_historyMod.F90(373): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"DX", &
------------------^
../core/LDT_historyMod.F90(375): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"DY", &
------------------^
../core/LDT_historyMod.F90(379): error #6285: There is no matching specific subroutine for this generic subroutine call.   [LDT_VERIFY]
             call LDT_verify(nf90_put_att(ftn,NF90_GLOBAL,"MAP_PROJECTION", &
------------------^
/tmp/ifortNjsWqw.i90(3396): catastrophic error: Too many errors, exiting
compilation aborted for ../core/LDT_historyMod.F90 (code 1)
make: *** [LDT_historyMod.o] Error 1

**[ERR] Compile failed**
---------------------------

[ERR] Compile failed

My env variables path are as follows:

module load intel/2020.4.304

export LDT_DIR=/projects/iist01/parent/LIS/depend

export LDT_FC=mpiifort
export LDT_ARCH=linux_ifc
export LDT_CC=mpiicc

export LDT_MODESMF=/projects/iist01/parent/LIS/depend/esmf/esmf/mod/modO/Linux.intel.64.intelmpi.default
export LDT_LIBESMF=/projects/iist01/parent/LIS/depend/esmf/esmf/lib/libO/Linux.intel.64.intelmpi.default
export LDT_OPENJPEG=/projects/iist01/parent/LIS/depend
export LDT_ECCODES=/projects/iist01/parent/LIS/depend
export LDT_NETCDF=/projects/iist01/parent/LIS/depend
export LDT_HDF4=/projects/iist01/parent/LIS/depend
export LDT_HDFEOS=/projects/iist01/parent/LIS/depend
export LDT_HDF5=/projects/iist01/parent/LIS/depend

export LD_LIBRARY_PATH=$LDT_HDF5/lib:$LDT_LIBESMF:$LDT_NETCDF/lib:$LDT_ECCODES/lib:$LDT_HDFEOS/lib:$LDT_HDF4/lib:$LD_LIBRARY_PATH

My configure.ldt is:
C              = mpiifort
FC77            = mpiifort
LD              = mpiifort
CC              = mpiicc
AR              = ar
MOD_ESMF        = /projects/iist01/parent/LIS/depend/esmf/esmf/mod/modO/Linux.intel.64.intelmpi.default
LIB_ESMF        = /projects/iist01/parent/LIS/depend/esmf/esmf/lib/libO/Linux.intel.64.intelmpi.default
INC_JPEG2000      = /projects/iist01/parent/LIS/depend/include/
LIB_JPEG2000      = /projects/iist01/parent/LIS/depend/lib64/
INC_ECCODES     = /projects/iist01/parent/LIS/depend/include/
LIB_ECCODES     = /projects/iist01/parent/LIS/depend/lib/
INC_NETCDF      = /projects/iist01/parent/LIS/depend/include/
LIB_NETCDF      = /projects/iist01/parent/LIS/depend/lib/
INC_HDF4        = /projects/iist01/parent/LIS/depend/include/
LIB_HDF4        = /projects/iist01/parent/LIS/depend/lib/
INC_HDF5        = /projects/iist01/parent/LIS/depend/include/
LIB_HDF5        = /projects/iist01/parent/LIS/depend/lib/
INC_HDFEOS      = /projects/iist01/parent/LIS/depend/include/
LIB_HDFEOS      = /projects/iist01/parent/LIS/depend/lib/
INC_FORTRANGIS1 =
INC_FORTRANGIS2 =
LIB_FORTRANGIS  =
INC_LIBGEOTIFF  =
LIB_LIBGEOTIFF  =
LIB_GDAL        =
CFLAGS          = -c  -DIFC
FFLAGS77        = -c -O2 -nomixed_str_len_arg -names lowercase -convert big_endian -assume byterecl  -DSPMD  -DIFC -I$(MOD_ESMF) -DUSE_INCLUDE_MPI -I$(INC_ECCODES) -I$(INC_NETCDF)  -I$(INC_HDFEOS)  -I$(INC_HDF4)  -I$(INC_HDF5)
FFLAGS          =  -c -O2 -u -traceback -fpe0  -nomixed_str_len_arg -names lowercase -convert big_endian -assume byterecl  -DSPMD -DIFC -I$(MOD_ESMF) -DUSE_INCLUDE_MPI -I$(INC_ECCODES) -I$(INC_NETCDF)  -I$(INC_HDFEOS)  -I$(INC_HDF4)  -I$(INC_HDF5)
LDFLAGS         =  -L$(LIB_ESMF) -lesmf -lstdc++ -limf -lm -lrt -lz -lmpi -L$(LIB_ECCODES) -leccodes_f90 -leccodes -L$(LIB_JPEG2000) -ljasper -L$(LIB_NETCDF) -lnetcdff -lnetcdf -L$(LIB_HDFEOS) -lhdfeos -lGctp -L$(LIB_HDF4) -lmfhdf -ldf -ljpeg -lz  -L$(LIB_HDF5) -lhdf5_fortran -lhdf5_hl -lhdf5 -ldl

Please Help!

**Comment by Bimingm** [comment]
Hello, did you resolve this problem? May I ask you a question?

**Comment by jvgeiger** [comment, marked as answer]
Hello,

```
../core/LDT_historyMod.F90(56): error #7013: This module file was not generated by any release of this compiler. [NETCDF]
use netcdf
```

suggests that you used one compiler to install NetCDF, but you used a different compiler to compile LDT.  You must make sure to use the same Fortran compiler for both installing the dependencies and compiling LISF.

**Reply by laurizioA** [reply]
@jvgeiger Is there a way to identify the NETCDF compiler and indicate the same for the LDT installation?

I have this probelam, but I don't know how to identify the NETCDF compiler.

**Reply by sudarsan3010** [reply]
nc-config --all
(check in the fc flag)
